synapse/types/__init__.py

```diff
@@ -327,6 +331,28 @@ class UserID(DomainSpecificString):
 
     SIGIL = "@"
 
+    def validate(self, allow_historical_mxids: Optional[bool] = False) -> bool:
```

Comment:
We already a) already do some basic validation on initialisation, and b) have an `is_valid`  function on `DomainSpecificString`. I'd be tempted to either a) move this check into `is_valid`, or b) name this something like `is_valid_user_id` (and have it as a staticmethod)

Reply:
Thanks for the suggestion, I manage to do something not too ugly using `kwargs` to be able to pass through `allow_historical_mxids` to `is_valid`.